process-document/src/main/java/com/ritense/processdocument/service/impl/DocumentDefinitionProcessLinkServiceImpl.java

```diff
@@ -41,24 +43,22 @@ public DocumentDefinitionProcessLinkServiceImpl(DocumentDefinitionProcessLinkRep
     }
 
     @Override
-    public DocumentDefinitionProcess getDocumentDefinitionProcess(String documentDefinitionName) {
-        var link= documentDefinitionProcessLinkRepository.findByIdDocumentDefinitionName(documentDefinitionName);
+    public List<DocumentDefinitionProcess> getDocumentDefinitionProcess(String documentDefinitionName) {
+        var links = documentDefinitionProcessLinkRepository.findAllByIdDocumentDefinitionName(documentDefinitionName);
 
-        if (link.isPresent()) {
+        return links.stream().map(link -> {
             var processDefinition = repositoryService.createProcessDefinitionQuery()
-                .processDefinitionKey(link.get().getId().getProcessDefinitionKey())
+                .processDefinitionKey(links.get(0).getId().getProcessDefinitionKey())
```

Comment:
Aren't you always still getting the first entry in the list this way? Then what's the point of getting a list?

Reply:
Changed